src/pubsub/ua_pubsub_ns0.c

```diff
@@ -847,10 +847,21 @@ addWriterGroupAction(UA_Server *server,
     writerGroupConfig.writerGroupId = writerGroupDataType->writerGroupId;
     writerGroupConfig.enabled = writerGroupDataType->enabled;
     writerGroupConfig.priority = writerGroupDataType->priority;
+    //TODO remove hard coded messageSettings
+    writerGroupConfig.messageSettings.encoding = UA_EXTENSIONOBJECT_DECODED;
+    writerGroupConfig.messageSettings.content.decoded.type = &UA_TYPES[UA_TYPES_UADPWRITERGROUPMESSAGEDATATYPE];
+    UA_UadpWriterGroupMessageDataType *writer_group_message = UA_UadpWriterGroupMessageDataType_new();
+    writer_group_message->networkMessageContentMask = (UA_UadpNetworkMessageContentMask)(UA_UADPNETWORKMESSAGECONTENTMASK_PUBLISHERID |
+    UA_UADPNETWORKMESSAGECONTENTMASK_GROUPHEADER |
+    UA_UADPNETWORKMESSAGECONTENTMASK_WRITERGROUPID |
+    UA_UADPNETWORKMESSAGECONTENTMASK_PAYLOADHEADER);
+    writerGroupConfig.messageSettings.content.decoded.data = writer_group_message;
     //TODO remove hard coded UADP
     writerGroupConfig.encodingMimeType = UA_PUBSUB_ENCODING_UADP;
     //ToDo transfer all arguments to internal WGConfiguration
     retVal |= UA_Server_addWriterGroup(server, *objectId, &writerGroupConfig, &generatedId);
+    retVal |= UA_Server_setWriterGroupOperational(server, generatedId);
```

Comment:
You're right, there is currently no option to set the WriterGroup operational over the information model. According to the standard the WriterGroup, respectively any type of Group, must contain an object called state (Part 14 9.1.6.2). These objects contain methods to enable and disable the Group. The added message settings should be set by the addDataSetWriter method (6.2.3.5). I've noted the missing configuration options and will fix them in the next weeks. For sure, you're also welcome to extend this PR and add the missing parts according to the OPC UA PubSub standard.